lib/spack/spack/patch.py

```diff
@@ -32,7 +32,14 @@ def apply_patch(stage, patch_path, level=1, working_dir='.'):
         working_dir (str): relative path *within* the stage to change to
             (default '.')
     """
-    patch = which("patch", required=True)
+    git_utils_path = os.environ.get('PATH', '')
+    if os.name == 'nt':
+        git = which_string('git', required=True)
+        git_root = os.path.dirname(git).split('/')[:-1]
+        git_root.extend(['usr', 'bin'])
+        git_utils_path = os.sep.join(git_root)
+
+    patch = which("patch", required=True, path=git_utils_path)
```

Comment:
These utilities are not traditionally on a Windows user's `PATH` as these are tools vendored by Git's internal copy of Bash. The `PATH` is typically set to `C:\Program Files (x86)\Git` or similar.

Reply:
Wow, that makes a lot of sense. There are absolutely ways to install git on Windows that do not include patch, as that component is vendored as a part of [Git for Windows](https://gitforwindows.org/) which is Git's support on Windows. Git can be built without the Git for Windows project i.e. from source or installed via chocolatey (I have no idea what they do), but I believe we decided that Spack required the Git for Windows installation, as we rely on other features, like Git Bash (and it's underlying sh support). Again I was not here for this initial discussion, so just to verify, is that more or less the gist of it @BetsyMcPhail? Otherwise, there are other ports of `patch` to Windows that I would be happy to retrieve and support via Spack package.

Reply:
> I believe we decided that Spack required the Git for Windows installation, 

To be clear: this is totally fine with me but I mainly think it's worth mentioning in a `TODO`-style comment that we should allow for other options in the future (e.g. I wasn't looking for actually supporting other means of locating patch in this PR).

Reply:
Ok, that's fine with me then.

Reply:
We do assume that Git for Windows... a comment specifying this would be helpfule

Reply:
I'll add it to the getting started guide. I think it may be best served there